### DISCUSSION TITLE
[Help Needed] Localization

### DISCUSSION THREAD

**Discussion by ApolloZhu** [ApolloZhu/Dynamic-Dark-Mode#86] [General]
If you'd like to translate this app into other languages so people who don't understand English can enjoy a productive life, please join as a translator on Crowdin: 

[**>>> Let's Do It**](https://crowdin.com/project/dynamic-dark-mode/invite?d=5545k4f6m6k59393g48373u2i473q4i4l4l4j4s223f4n49353e3n45353d30343a31353t663q483a38373)

----

- [x] English
- [x] 简体中文（Chinese, Simplified）
- [x] Deutsch (German) - Thanks to @LukasLen 
- [x] Русский язык (Russian) - Thanks to @pankova and @nevack
- [x] Bahasa Indonesia (Indonesian) - Thanks to @hndr91 
- [x] le français (French) - Thanks to @Ethenyl and @flayks
- [x] Español (Spanish) - Thanks to @nununoisy
- [ ] Türkçe (Turkish) - In progress by @kahyalar
- [ ] Esperanto - Almost done by @Vince14Genius
- [ ] 한국어 (Korean) - @anaclumos will continue after his final exams
- [ ] 日本語 (Japanese) - @SaziumR has started
- [ ] ...

Few things to be aware of:
- Translation of "Dark Mode" and other terms used by Apple should match those used in macOS.
- When translating the app name, think about what you'll search in Google (or other search engine) when you try to find a software that toggles dark mode automatically. That's why it's English name is "Dynamic Dark Mode".
- When in doubt, don't translate. Instead, you should research, learn new things, then translate.

I'll try my best to answer your other questions. Thank you for your support in advance.

**Comment by hndr91** [comment]
May I help sir? Which file should be localized?

**Comment by ApolloZhu** [comment]
Hello again, @hndr91 ! Which language can you translate this to? I assume it's Indonesian?

**Comment by ApolloZhu** [comment]
@hndr91 This is the XLIFF Localization File for Indonesian: [id.txt](https://github.com/ApolloZhu/Dynamic-Dark-Mode/files/2453760/id.txt). Rename it to `id.xliff` so you can open it in an xliff editor.

- Do you want to send a pull request?
  - Yes: Do you have Xcode?
    - Yes: Translate the xliff file, and use `Editor` > `Import Localizations...` to import the finished version
    - No: Translate all 4 .strings file under the 2 `id.lproj` folders.
  - No:  When you are done translating the xliff file, just send the finished xliff file back to me.

**Comment by hndr91** [comment]
@ApolloZhu I will try it. I got mac, but I didn't update my OS, so i'll do it manually

**Comment by pankova** [comment]
Hi @ApolloZhu! Do you need Russian localization? I can help

**Comment by ApolloZhu** [comment]
@pankova Yes and thank you very much! The process is the same as I described above, but please use this file for Russian: [ru.txt](https://github.com/ApolloZhu/Dynamic-Dark-Mode/files/2458318/ru.txt)

**Comment by hndr91** [comment]
@ApolloZhu hi, I'm working on Indonesia translation now. I figure out that you use upper case letter in the beginning of every word. Is that the standard?

**Comment by ApolloZhu** [comment]
Those in title case are for titles (and in preferences pane since they look nice, at least to me). I’m not sure if Indonesian has the concept of title case too, so anything you think it’s appropriate and aesthetically pleasing is acceptable.

**Comment by pankova** [comment]
@ApolloZhu @hndr91 in Russian, for instance, there isn't the concept of title case for titles :)

**Comment by hndr91** [comment]
@ApolloZhu @pankova Ok, I'll check it again

**Comment by damien-rivet** [comment]
Hi @ApolloZhu,

I will work on the French translation, opening a PR soon.

**Comment by ApolloZhu** [comment]
@SaziumR thank you very much!

**Comment by ApolloZhu** [comment]
Hello everyone! I'm moving translation to Crowdin so we can *track progress* and *update translations* easier. You may now **join as a translator** using this link below:

https://crowdin.com/project/dynamic-dark-mode/invite?d=5545k4f6m6k59393g48373u2i473q4i4l4l4j4s223f4n49353e3n45353d30343a31353t663q483a38373

**Comment by kahyalar** [comment]
Hey @ApolloZhu,

I can work on Turkish translations if you are looking for.

Cheers!

**Comment by ApolloZhu** [comment]
@kahyalar Thanks!

- If you want to earn your pull request for Hacktoberfest
  - there are some instructions in this thread that might be helpful on how to do this
- otherwise you can use crowdin, I just added the option for you to translate to Turkish
  - You can also probably export xliff file from this site and import that file into Xcode and make a PR

Anyways, let me know if you have any other questions

**Comment by besilva** [comment]
Hello @ApolloZhu I can translate to Brazilian Portuguese if you are looking for.

**Comment by ApolloZhu** [comment]
@besilva I have enabled Brazilian Portuguese in Crowdin. Thank you very much!

**Comment by Vistaus** [comment]
Hi @ApolloZhu ! Can you add Dutch on Crowdin so that I can translate the app to Dutch? :)

**Comment by ApolloZhu** [comment]
@Vistaus added! Thanks :)

**Comment by Vistaus** [comment]
@ApolloZhu Thanks! :) I sent a join request on Crowdin to join the newly added Dutch translation.

**Comment by ApolloZhu** [comment]
@Vistaus  request accepted

**Comment by Vistaus** [comment]
@ApolloZhu Great! I'll start translating right away! :-)
